pkg/server/api/agent/v1/service.go

```diff
@@ -47,7 +55,52 @@ func (s *Service) DeleteAgent(ctx context.Context, req *agent.DeleteAgentRequest
 }
 
 func (s *Service) BanAgent(ctx context.Context, req *agent.BanAgentRequest) (*empty.Empty, error) {
-	return nil, status.Error(codes.Unimplemented, "method not implemented")
+	log := rpccontext.Logger(ctx)
+
+	if req.Id == nil {
+		log.Error("Invalid request: ID is nil")
+		return nil, status.Error(codes.InvalidArgument, "ID is nil")
+	}
+
+	id, err := spiffeid.New(req.Id.TrustDomain, req.Id.Path)
+	if err != nil {
+		log.WithError(err).Error("Invalid request: invalid SPIFFE ID")
+		return nil, status.Errorf(codes.InvalidArgument, "invalid SPIFFE ID: %v", err)
+	}
+
+	log = log.WithField(telemetry.SPIFFEID, id.String())
+
+	if !strings.Contains(id.Path(), "spire/agent/") {
```

Comment:
do we want to put this hardcoded string constant somewhere else? (maybe the spiffeid package)

Reply:
Sounds good! Not sure if `spiffeid` package is a good fit because this string is related to how SPIRE handles agent IDs and it is not part of the SPIFFE standard itself. I'll look for a place for it... perhaps it could be in`idutil` package.

Reply:
mmm I dont think Containis is a good choice here, we must verify it has a prefix:
```
strings.HasPrefix(id.Path(), "/spire/agent/")
```

what do you think about creating a method to verify an spiffeID is agent? maybe on api.go? 
something like:
```
func IsAgent(id spiffeid.Id) bool {
   strings.HasPrefix(id.Path(), "/spire/agent/")
}
```

Reply:
Sounds good! I noticed a [similar function](https://github.com/spiffe/spire/blob/62ff83babd4a0738ca333c1aed04958433281186/pkg/common/idutil/spiffeid.go#L144) already exists in the `idutil` package. Perhaps we can export it and reuse it instead of making a new one?